### ISSUE TITLE
[Issue]: GPU training suddenly produces NaN losses, CPU works fine

### ISSUE THREAD

**Issue by Doc415** [ROCm/TheRock#2662]
### Problem Description

Hi,

Until a few days ago everything was working normally. Suddenly, all my PyTorch trainings started producing NaN losses when running on GPU (ROCm). This happens across different models and training scripts.

Key points:

Same code and data work fine on CPU

NaNs appear only on GPU

Happens very early in training

I reinstalled AMD drivers, ROCm, Python, and PyTorch from scratch

Issue still persists

No intentional code changes before this started.

PyTorch version: 2.11.0a0+rocm7.11.0a20251217

ROCm (CUDA backend) available: True

Number of GPUs: 1

GPU name: AMD Radeon RX 7600

ROCm (HIP) runtime version: 7.2.53150

PyTorch build configuration:

PyTorch built with:

- C++ Version: 201703

- clang 22.0.0

- MSVC 194435222

- Intel(R) MKL-DNN v3.7.1 (Git Hash 8d263e693366ef8db40acc569cc7d8edf644556d)

- OpenMP 202011

- LAPACK is enabled (usually provided by MKL)

- CPU capability usage: AVX2

- HIP Runtime 7.2.53150

- MIOpen 3.5.1

- Build settings: BLAS_INFO=open, BUILD_TYPE=Release, COMMIT_SHA=f814614e6ff0833f82a4a29a5a14b9fa7287e8ab, CXX_COMPILER=C:/home/runner/_work/_tool/Python/3.13.11/x64/Lib/site-packages/_rocm_sdk_devel/lib/llvm/bin/clang-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /EHsc /Zc:__cplusplus /bigobj /FS /utf-8 -DUSE_PTHREADPOOL -DNDEBUG -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE /wd4624 /wd4068 /wd4067 /wd4267 /wd4661 /wd4717 /wd4244 /wd4804 /wd4273, LAPACK_INFO=open, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, TORCH_VERSION=2.11.0, USE_CUDA=OFF, USE_CUDNN=OFF, USE_CUSPARSELT=OFF, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_GLOO=OFF, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_ROCM=ON, USE_ROCM_KERNEL_ASSERT=OFF, USE_XCCL=OFF, USE_XPU=OFF,

When I felt back to rockm 7.10 with pytorch 2.10 , evrything works fine.

[Crosspost to more communities](https://www.reddit.com/submit/?source_id=t3_1pqvnxb)

### Operating System

Windows 10

### CPU

AMD Ryzen 5 5600

### GPU

Radeon RX 7600

### ROCm Version

rocm7.11.0a20251217

### Steps to Reproduce

```
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import Dataset, DataLoader
from torchvision import models

device = "cuda" if torch.cuda.is_available() else "cpu"
print("Device:", device)

class SyntheticDataset(Dataset):
    def __init__(self, n=50):
        self.x = torch.randn(n, 3, 224, 224)
        self.y = torch.randint(0, 2, (n,))

    def __len__(self):
        return len(self.x)

    def __getitem__(self, i):
        return self.x[i], self.y[i]

dataset = SyntheticDataset(50)
loader = DataLoader(dataset, batch_size=8, shuffle=True)

model = models.resnet18(weights=None)
model.fc = nn.Linear(model.fc.in_features, 2)
model = model.to(device)

criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=1e-3)

model.train()
epochs = 2

for epoch in range(epochs):
    print(f"\nEpoch {epoch+1}/{epochs}")
    for i, (x, y) in enumerate(loader):
        x = x.to(device)
        y = y.to(device)

        optimizer.zero_grad()
        logits = model(x)
        loss = criterion(logits, y)
        loss.backward()
        optimizer.step()

        print(f"Epoch {epoch+1} | Batch {i+1}/{len(loader)} | Loss: {loss.item():.6f}")
```

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

Device: cuda

Epoch 1/2
Epoch 1 | Batch 1/7 | Loss: 0.000000
Epoch 1 | Batch 2/7 | Loss: 0.698797
Epoch 1 | Batch 3/7 | Loss: 0.695984
Epoch 1 | Batch 4/7 | Loss: 0.693200
Epoch 1 | Batch 5/7 | Loss: 0.690786
Epoch 1 | Batch 6/7 | Loss: 0.690560
Epoch 1 | Batch 7/7 | Loss: 0.693210

Epoch 2/2
Epoch 2 | Batch 1/7 | Loss: nan
Epoch 2 | Batch 2/7 | Loss: nan
Epoch 2 | Batch 3/7 | Loss: nan
Epoch 2 | Batch 4/7 | Loss: nan
Epoch 2 | Batch 5/7 | Loss: nan
Epoch 2 | Batch 6/7 | Loss: nan
Epoch 2 | Batch 7/7 | Loss: 0.000000

**Comment by adityas-amd** [comment]
@Doc415 Thank you for reporting, similar to https://github.com/ROCm/TheRock/issues/2510, are you finding the nan on all MIOPEN_MODES? could you please share running the script with `MIOPEN_ENABLE_LOGGING=1` `MIOPEN_ENABLE_LOGGING_CMD=1` `MIOPEN_LOG_LEVEL=6`

**Comment by Doc415** [comment]
> [@Doc415](https://github.com/Doc415) Thank you for reporting, similar to [#2510](https://github.com/ROCm/TheRock/issues/2510), are you finding the nan on all MIOPEN_MODES? could you please share running the script with `MIOPEN_ENABLE_LOGGING=1` `MIOPEN_ENABLE_LOGGING_CMD=1` `MIOPEN_LOG_LEVEL=6`

Thanks for the follow-up.

Yes — similar to #2510, I observed NaNs across the different MIOPEN_MODES I tried; the outcome did not change between modes. In a couple of modes the system became unresponsive (locked).

At the moment I’m not able to rerun the script with:
MIOPEN_ENABLE_LOGGING=1 MIOPEN_ENABLE_LOGGING_CMD=1 MIOPEN_LOG_LEVEL=6

I had to roll back my environment from ROCm 7.11 + PyTorch 2.11 to ROCm 7.10 + PyTorch 2.10 (I removed the newer stack), and I don’t have the 7.11/2.11 setup available right now to reproduce and capture the logs.

**Comment by adityas-amd** [comment]
@Doc415 I reproduced with `pip install --index-url=https://rocm.nightlies.amd.com/v2-staging/gfx110X-all/ --pre torch==2.11.0a0+rocm7.11.0a20260119 torchvision torchaudio`
Seems to be fixed, coud you please install above and try again. my results are on first attempt (no kernel restarts)
<img width="1612" height="949" alt="Image" src="https://github.com/user-attachments/assets/ee90716a-09de-4bba-a471-c841303210fe" />

**Comment by Doc415** [comment]
> [@Doc415](https://github.com/Doc415) I reproduced with `pip install --index-url=https://rocm.nightlies.amd.com/v2-staging/gfx110X-all/ --pre torch==2.11.0a0+rocm7.11.0a20260119 torchvision torchaudio` Seems to be fixed, coud you please install above and try again. my results are on first attempt (no kernel restarts) <img alt="Image" width="1612" height="949" src="https://private-user-images.githubusercontent.com/195787814/537761473-ee90716a-09de-4bba-a471-c841303210fe.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg5MjE1NDYsIm5iZiI6MTc2ODkyMTI0NiwicGF0aCI6Ii8xOTU3ODc4MTQvNTM3NzYxNDczLWVlOTA3MTZhLTA5ZGUtNGJiYS1hNDcxLWM4NDEzMDMyMTBmZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyMFQxNTAwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNGMyNmViNTRhYTJiMzJmNTA0MTE0OTE4MjIxMjIzNDdhYmVhNzcwMDNmZWNhNjAyOTY3NGFmYjMxMmJhYzFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.A364zmgOTUu7f7pcZlm4iWFBKfGlzHEfTib2Ws3B8DA">

Thanks for the fix. I retested on ROCm and the issue is resolved now on my side.
I’ll close this issue.

Device: cuda
PyTorch version: 2.11.0a0+rocm7.11.0a20260119
ROCm available (CUDA backend): None
HIP (ROCm) version: 7.2.53150

Epoch 1/2
Epoch 1 | Batch 1/7 | Loss: 1.030231
Epoch 1 | Batch 2/7 | Loss: 0.696656
Epoch 1 | Batch 3/7 | Loss: 0.667086
Epoch 1 | Batch 4/7 | Loss: 1.487400
Epoch 1 | Batch 5/7 | Loss: 1.046839
Epoch 1 | Batch 6/7 | Loss: 0.644602
Epoch 1 | Batch 7/7 | Loss: 0.710099

Epoch 2/2
Epoch 2 | Batch 1/7 | Loss: 0.423326
Epoch 2 | Batch 2/7 | Loss: 0.256186
Epoch 2 | Batch 3/7 | Loss: 0.092315
Epoch 2 | Batch 4/7 | Loss: 0.251253
Epoch 2 | Batch 5/7 | Loss: 3.629720
Epoch 2 | Batch 6/7 | Loss: 0.005394
Epoch 2 | Batch 7/7 | Loss: 0.278476
